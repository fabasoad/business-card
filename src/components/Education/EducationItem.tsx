import * as React from 'react'
import { useTranslation } from 'react-i18next'
import DateLocale from '../controls/DateLocale'

interface EducationItemProps {
  from: number
  to: number
  title: string
}

export default function EducationItem(props: EducationItemProps) {
  const { t } = useTranslation()
  return (
    <>
      <div className="timeline-image">
        <h4><DateLocale year={props.to} />-<br /><DateLocale year={props.from} /></h4>
      </div>
      <div className="timeline-panel">
        <div className="timeline-heading">
          <h4>{t('business-card-education-university-title')}</h4>
          <h4 className="subheading">{props.title}</h4>
        </div>
        <div className="timeline-body">
          <p>{t('business-card-education-university-speciality')}</p>
        </div>
      </div>
    </>
  )
}
