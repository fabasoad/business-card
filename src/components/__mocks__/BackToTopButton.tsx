import * as React from 'react'

export default function BackToTopButton() {
  return (
    <div data-testid="BackToTopButton"></div>
  )
}
