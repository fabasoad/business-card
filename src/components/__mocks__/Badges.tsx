import * as React from 'react'

export default function Badges() {
  return (
    <div data-testid="Badges"></div>
  )
}
