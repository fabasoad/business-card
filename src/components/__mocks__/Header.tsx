import * as React from 'react'

export default function Header() {
  return (
    <div data-testid="Header"></div>
  )
}
