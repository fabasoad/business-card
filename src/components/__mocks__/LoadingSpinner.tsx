import * as React from 'react'
import { LoadingSpinnerProps } from '../LoadingSpinner'

export default function LoadingSpinner({ background, color }: LoadingSpinnerProps) {
  return (
    <div data-testid={`LoadingSpinner-${background}-${color}`}></div>
  )
}
