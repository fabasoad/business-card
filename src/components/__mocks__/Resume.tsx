import * as React from 'react'

export default function Resume() {
  return (
    <div data-testid="Resume"></div>
  )
}
