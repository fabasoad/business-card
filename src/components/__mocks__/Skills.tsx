import * as React from 'react'

export default function Skills() {
  return (
    <div data-testid="Skills"></div>
  )
}
