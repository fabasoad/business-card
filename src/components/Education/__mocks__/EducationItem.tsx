import * as React from 'react'
import { EducationItemProps } from '../EducationItem'

export default function EducationItem({
  universityKey, universityName, location, img, fromYear, toYear
}: EducationItemProps) {
  return (
    <div data-testid={`EducationItem-${universityKey}-${universityName}-${location}-${fromYear}-${toYear}-${img}`}>
    </div>
  )
}
