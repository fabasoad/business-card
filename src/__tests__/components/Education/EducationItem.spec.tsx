/// <reference types="jest" />
import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import EducationItem from '../../../components/Education/EducationItem'
import { useTranslation } from '../../__mocks__/react-i18next'

let tMock

beforeAll(() => {
  tMock = useTranslation().t
})

test('should render EducationItem correctly', () => {
  const wrapper: ShallowWrapper = shallow(<EducationItem
    from={2019}
    to={2020}
    title="test-title"
  />)
  expect(wrapper).toMatchSnapshot()
  expect(tMock).toHaveBeenCalledTimes(2)
  expect(tMock).toHaveBeenCalledWith('business-card-education-university-title')
  expect(tMock).toHaveBeenCalledWith('business-card-education-university-speciality')
})
