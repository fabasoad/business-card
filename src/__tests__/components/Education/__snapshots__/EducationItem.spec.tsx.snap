// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EducationItem correctly 1`] = `
<Fragment>
  <div
    className="timeline-image"
  >
    <h4>
      <Connect(DateLocale)
        year={2020}
      />
      -
      <br />
      <Connect(DateLocale)
        year={2019}
      />
    </h4>
  </div>
  <div
    className="timeline-panel"
  >
    <div
      className="timeline-heading"
    >
      <h4>
        business-card-education-university-title
      </h4>
      <h4
        className="subheading"
      >
        test-title
      </h4>
    </div>
    <div
      className="timeline-body"
    >
      <p>
        business-card-education-university-speciality
      </p>
    </div>
  </div>
</Fragment>
`;
